kuksa-sdk/src/main/kotlin/org/eclipse/kuksa/MultiListener.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.eclipse.kuksa
+
+/**
+ * Generic Listener interface, to support multiple listeners.
+ */
+class MultiListener<T> {
```

Comment:
Good point, this was an experiment of me, to a) reduce the number of "registerListenerX" and "unregisterListenerX" methods inside DataBrokerConnection (each Listener obviously will come with two methods) and b) to share the underlying logic (e.g. use a set to prevent multiple registrations of the same listener) using a common interface, why do I need to always re-implement registration / unregistration logic even though it's the same logic for all listeners? 

If you don't like the syntax dataBrokerConnection.disconnectListener.registerListener() (which really might feel a bit unusual) then we can make the disconnectListener private and expose methods for (un)registerListener in DataBrokerConnection